### ISSUE TITLE
Implement TLS support

### ISSUE THREAD

**Issue by senseibaka** [DataAction/AdoNetCore.AseClient#98]
Investigate if it's possible, and what versions of TLS are supported (hopefully at least 1.2)

**Comment by senseibaka** [comment]
### Server commands

```
-- Get the server name (for assigning a certificate)
select @@servername
-- Check what your server is licenced for with
sp_lmconfig
-- Enable SSL
sp_configure 'enable ssl', 1
-- Use a certificate
sp_ssladmin 'addcert', 'PATH/TO/Certificate', 'password'
```

### Creating/configuring a certificate (OpenSSL)

1. Generate a CSR, but make sure you've specified a subjectAltName of `DNS:<result of @@servername>`
2. Send off the CSR to your favourite CA
3. Process the CA's response
4. Copy your cert over to the database server's filesystem
5. Run commands above to enable SSL (might need to restart the server)

**Comment by c-j-hughes** [comment]
Fixed by #122 .
